<robot name="test_robot">


  <material name="blue">
    <color rgba="0.3765 0.5647 0.101 1"/>
  </material>

  <material name="gray">
    <color rgba="0.0 0.0 0.0 1"/>
  </material>
  
  <link name="base_link">
    <visual>
      <geometry>
        <cylinder length="0.5" radius="0.05"/>
      </geometry>
      <material name="blue"/>
    </visual>
  </link>

  <link name="link2">
    <visual>
      <geometry>
        <cylinder length="0.5" radius="0.05"/>
      </geometry>
      <origin xyz=" 0 0 0.25" rpy = "0 0 0" />
      <material name="gray"/>
    </visual>
  </link>

  <link name="link3">
    <visual>
      <geometry>
        <cylinder length="0.5" radius="0.05"/>
      </geometry>
      <material name="blue"/>
      <origin xyz=" 0 0 0.25" rpy = "0 0 0" />
    </visual>
  </link>

  <link name="link4">
    <visual>
      <geometry>
        <cylinder length="0.5" radius="0.05"/>
      </geometry>
    <material name="gray"/>
    <origin xyz=" 0 0 0.25" rpy = "0 0 0" />
    </visual>
  </link>

  <link name="link5">
    <visual>
      <geometry>
        <cylinder length="0.5" radius="0.05"/>
      </geometry>
      <material name="blue"/>
      <origin xyz=" 0 0 0.25" rpy = "0 0 0" />
    </visual>
  </link>

    <link name="link6">
    <visual>
      <geometry>
        <cylinder length="0.5" radius="0.05"/>
      </geometry>
      <material name="gray"/>
      <origin xyz=" 0 0 0.25" rpy = "0 0 0" />
    </visual>
  </link>

  <link name="link7">
    <visual>
      <geometry>
        <cylinder length="0.5" radius="0.05"/>
      </geometry>
      <material name="blue"/>
      <origin xyz=" 0 0 0.25" rpy = "0 0 0" />
    </visual>
  </link>

  <link name="link8">
    <visual>
      <geometry>
        <cylinder length="0.5" radius="0.05"/>
      </geometry>
      <material name="gray"/>
      <origin xyz=" 0 0 0.25" rpy = "0 0 0" />
    </visual>
  </link>
  
  <link name="link9">
    <visual>
      <geometry>
        <cylinder length="0.5" radius="0.05"/>
      </geometry>
      <material name="blue"/>
      <origin xyz=" 0 0 0.25" rpy = "0 0 0" />
    </visual>
  </link>

    <link name="link10">
    <visual>
      <geometry>
        <cylinder length="0.5" radius="0.05"/>
      </geometry>
      <material name="gray"/>
      <origin xyz=" 0 0 0.25" rpy = "0 0 0" />
    </visual>
  </link>

  <joint name="joint1" type="continuous">
    <parent link="base_link"/>
    <child link="link2"/>
    <origin xyz=" 0 0 0.25" rpy = "0 0 0" />
    <axis xyz="1 0 0" />
  </joint>

  <joint name="joint2" type="continuous">
    <parent link="link2"/>
    <child link="link3"/>
    <origin xyz=" 0 0 0.5" rpy = "0 0 0" />
    <axis xyz="0 1 0" />
  </joint>

  <joint name="joint3" type="continuous">
    <parent link="link3"/>
    <child link="link4"/>
    <origin xyz=" 0 0 0.5" rpy = "0 0 0" />
    <axis xyz="1 0 0" />
  </joint>
  
  <joint name="joint4" type="continuous">
    <parent link="link4"/>
    <child link="link5"/>
    <origin xyz=" 0 0 0.5" rpy = "0 0 0" />
    <axis xyz="0 1 0" />
  </joint>
  
  <joint name="joint5" type="continuous">
    <parent link="link5"/>
    <child link="link6"/>
    <origin xyz=" 0 0 0.5" rpy = "0 0 0" />
    <axis xyz="1 0 0" />
  </joint>

  <joint name="joint6" type="continuous">
    <parent link="link6"/>
    <child link="link7"/>
    <origin xyz=" 0 0 0.5" rpy = "0 0 0" />
    <axis xyz="0 1 0" />
  </joint>
  
  <joint name="joint7" type="continuous">
    <parent link="link7"/>
    <child link="link8"/>
    <origin xyz=" 0 0 0.5" rpy = "0 0 0" />
    <axis xyz="1 0 0" />
  </joint>

  <joint name="joint8" type="continuous">
    <parent link="link8"/>
    <child link="link9"/>
    <origin xyz=" 0 0 0.5" rpy = "0 0 0" />
    <axis xyz="1 0 0" />
  </joint>

  <joint name="joint9" type="continuous">
    <parent link="link9"/>
    <child link="link10"/>
    <origin xyz=" 0 0 0.5" rpy = "0 0 0" />
    <axis xyz="1 0 0" />
  </joint>

</robot>